cmake/OpenCVCRTLinkage.cmake

```diff
@@ -2,47 +2,37 @@ if(NOT MSVC)
   message(FATAL_ERROR "CRT options are available only for MSVC")
 endif()
 
-#INCLUDE (CheckIncludeFiles)
+#if (${CMAKE_SYSTEM_NAME} MATCHES "WindowsStore" OR ${CMAKE_SYSTEM_NAME} MATCHES "WindowsPhone")
+#  set(WINRT TRUE)
 
-set(HAVE_WINRT FALSE)
-
-# search Windows Platform SDK
-message(STATUS "Checking for Windows Platform SDK")
-GET_FILENAME_COMPONENT(WINDOWS_SDK_PATH  "[HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Microsoft SDKs\\Windows\\v8.0;InstallationFolder]" ABSOLUTE CACHE)
-if(WINDOWS_SDK_PATH STREQUAL "")
-  set(HAVE_MSPDK FALSE)
-  message(STATUS "Windows Platform SDK 8.0 was not found")
-else()
-  set(HAVE_MSPDK TRUE)
+if (WINRT)
+  add_definitions(/DWINVER=_WIN32_WINNT_WIN8 /DNTDDI_VERSION=NTDDI_WIN8 /D_WIN32_WINNT=_WIN32_WINNT_WIN8)
 endif()
 
-#search for Visual Studio 11.0 install directory
-message(STATUS "Checking for Visual Studio 2012")
-GET_FILENAME_COMPONENT(VISUAL_STUDIO_PATH [HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\VisualStudio\\11.0\\Setup\\VS;ProductDir] REALPATH CACHE)
-if(VISUAL_STUDIO_PATH STREQUAL "")
-  set(HAVE_MSVC2012 FALSE)
-  message(STATUS "Visual Studio 2012 was not found")
-else()
-  set(HAVE_MSVC2012 TRUE)
-endif()
+# Removing LNK4075 warnings for debug WinRT builds
+# "LNK4075: ignoring '/INCREMENTAL' due to '/OPT:ICF' specification"
+# "LNK4075: ignoring '/INCREMENTAL' due to '/OPT:REF' specification"
+if(MSVC AND WINRT)
+  # Optional verification checks since we don't know existing contents of variables below
+  string(REPLACE "/OPT:ICF " "/OPT:NOICF " CMAKE_EXE_LINKER_FLAGS_DEBUG "${CMAKE_EXE_LINKER_FLAGS_DEBUG}")
+  string(REPLACE "/OPT:REF " "/OPT:NOREF " CMAKE_EXE_LINKER_FLAGS_DEBUG "${CMAKE_EXE_LINKER_FLAGS_DEBUG}")
+  string(REPLACE "/INCREMENTAL:YES " "/INCREMENTAL:NO " CMAKE_EXE_LINKER_FLAGS_DEBUG "${CMAKE_EXE_LINKER_FLAGS_DEBUG}")
+  string(REPLACE "/INCREMENTAL " "/INCREMENTAL:NO " CMAKE_EXE_LINKER_FLAGS_DEBUG "${CMAKE_EXE_LINKER_FLAGS_DEBUG}")
 
-try_compile(HAVE_WINRT_SDK
-  "${OpenCV_BINARY_DIR}"
-  "${OpenCV_SOURCE_DIR}/cmake/checks/winrttest.cpp")
```

Comment:
Is `winrttest.cpp` file used anywhere? Please remove if you haven't any plans for using it.